sklearn/tree/_criterion.pyx

```diff
@@ -446,42 +435,39 @@ cdef class ClassificationCriterion(Criterion):
                 if sample_weight != NULL:
                     w = sample_weight[i]
 
-                for k in range(n_outputs):
-                    label_index = (k * label_count_stride +
-                                   <SIZE_t> y[i * y_stride + k])
-                    label_count_left[label_index] += w
+                for k in range(self.n_outputs):
+                    label_index = (k * self.sum_stride +
+                                   <SIZE_t> y[i * self.y_stride + k])
+                    sum_left[label_index] += w
 
-                diff_w += w
+                self.weighted_n_left += w
+                self.weighted_n_right -= w
 
         else:
             self.reverse_reset()
 
             for p in range(end - 1, new_pos - 1, -1):
-                i = samples[p]
+                i = self.samples[p]
```

Comment:
Why did unpack `samples` above but not here? Is it just an omission and or is there a reason?